src/iarc7_motion/high_level_motion_controller.py

```diff
@@ -0,0 +1,213 @@
+                        done_cb=self._safety_task_complete_callback)
+                rospy.logwarn('motion planner attempting to execute safety land')
+                self._safety_land_requested = True
+
+            # if we have not seen a task yet
+            # we wait until now to start the task timeout timer
+            # because the action clients have not yet been initialized
+            if not self._initialized:
+                self._timer = rospy.Timer(self._task_timeout, self._recieve_task_timeout)
+                self._initialized = True
+
+            if self._task is None:
+                if self._action_server.has_new_task():
+                    new_task = self._action_server.get_new_task()
+
+                    if self._state_monitor.check_transition(new_task):
+                        self._task = new_task
+                        self._shutdown_timer()
+                        self._task_command_handler.new_task(new_task, self._get_transition())
+                    else: 
+                        rospy.logwarn('AI provided illegal task transition. Aborting requested task.')
+                        self._action_server.set_aborted()
+            else: 
+                if self._action_server.is_canceled():
+                    self._task_command_handler.cancel_task()
+
+                self._task_command_handler.run()
+                task_state = self._task_command_handler.get_state()
+
+                # handles state of task and HLM controller
+                if isinstance(task_state, task_states.TaskCanceled):
+                    self._action_server.set_canceled()
+                    rospy.logwarn('Task was canceled')
+                    self._task = None
+                elif isinstance(task_state, task_states.TaskAborted):
+                    rospy.logwarn('Task aborted with: %s', task_state.msg)
+                    self._action_server.set_aborted()
+                    self._task = None
+                elif isinstance(task_state, task_states.TaskFailed):
+                    rospy.logwarn('Task failed with: %s', task_state.msg)
+                    self._action_server.set_succeeded(False)
+                    self._task = None
+                elif isinstance(task_state, task_states.TaskDone):
+                    self._action_server.set_succeeded(True)
+                    self._task = None
+                else:
+                    assert isinstance(task_state, task_states.TaskRunning)
+
+                # as soon as we set a task to None, start timeout timer
+                if self._task is None:
+                    self._timer = rospy.Timer(self._task_timeout, self._recieve_task_timeout)
+                    self._timeout_vel_sent = False
+
+            rate.sleep()
+
+    # fills out the Intermediary State for the task
+    def _get_transition(self):
+        state = IntermediaryState() 
+        state.last_twist = self._task_command_handler.get_last_twist()
```

Comment:
It knows the actual velocity, but I thought it might be useful later to also know what the last requested twist was. 

Additionally, we need it for timeouts. Granted, the state monitor could listen to the same request topic as LLM but I wasn't sure if we wanted to do that. 

Maybe also a more robust timeout scheme is needed, one that looks at the actual drone's velocity and makes a request based off of that.